### PR TITLE
Remove regular_type from function with no complex dependencies

### PR DESCRIPTION
**Pull request by jfalcou**
The `regular`/`plain`and `musl` decorator are rather useless under the new way we wrote code.
This is a first PR to remove `regular` from a sensible chunk of functions, turning them into new callabels at the same time.
More PRs of this sort are gonna comes out before we can remove the decorator themselves.

This PR also contains drive-by refactoring and simplification, and a bunch of file deletions.

### CODE REVIEW

File: include/eve/module/core/regular/rotr.hpp [hunk 1]
```diff
@@ -1,45 +1,69 @@
...
 {
-namespace tag
-{
-  struct rotr_;
+  template<typename Options>
+  struct rotr_t : elementwise_callable<rotr_t, Options>
+  {
+    template<eve::integral_value T, eve::integral_value S>
+    requires(unsigned_scalar_value<element_type_t<T>>)
+    constexpr EVE_FORCEINLINE as_wide_as_t<T,S> operator()(T v, S s) const
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/rotr.hpp:24]: why constexpr

File: test/unit/module/math/expm1.cpp [hunk 2]
```diff
@@ -27,21 +24,17 @@ TTS_CASE_TPL("Check return types of expm1", eve::test::simd::ieee_reals)
...
   using eve::detail::map;
   using v_t = eve::element_type_t<T>;
 
   TTS_ULP_EQUAL(eve::expm1(a0), map([](auto e) -> v_t { return std::expm1(e); }, a0), 30);
   TTS_ULP_EQUAL(eve::expm1(a1), map([](auto e) -> v_t { return std::expm1(e); }, a1), 2);
-
-  TTS_ULP_EQUAL(
-      eve::pedantic(eve::expm1)(a0), map([](auto e) -> v_t { return std::expm1(e); }, a0), 30);
-  TTS_ULP_EQUAL(
-      eve::pedantic(eve::expm1)(a1), map([](auto e) -> v_t { return std::expm1(e); }, a1), 2);
```
Comment (DenisYaroshevskiy) [inline test/unit/module/math/expm1.cpp:44]: i don't get this removal

File: test/unit/module/core/ldexp.cpp [hunk 3]
```diff
@@ -25,52 +25,35 @@ TTS_CASE_TPL("Check return types of ldexp", eve::test::simd::ieee_reals)
...
+  TTS_EXPR_IS(ldexp[pedantic](T(), T())     , T   );
+  TTS_EXPR_IS(ldexp[pedantic](T(), v_t())   , T   );
+  TTS_EXPR_IS(ldexp[pedantic](v_t(), v_t()) , v_t );
+  TTS_EXPR_IS(ldexp[pedantic](v_t(), T())   , T   );
 };
 
-//==================================================================================================
-// ldexp  simd tests
-//==================================================================================================
-TTS_CASE_WITH("Check behavior of ldexps on floatings",
```
Comment (DenisYaroshevskiy) [inline test/unit/module/core/ldexp.cpp:54]: what's with this test?

File: include/eve/module/special/regular/rising_factorial.hpp [hunk 4]
```diff
@@ -14,11 +14,10 @@
...
 {
   template<typename Options>
-  struct rising_factorial_t : elementwise_callable<rising_factorial_t, Options, raw_option, pedantic_option, regular_option>
+  struct rising_factorial_t : elementwise_callable<rising_factorial_t, Options, raw_option, pedantic_option>
   {
-    template<eve::ordered_value I, eve::floating_ordered_value T>
-    EVE_FORCEINLINE constexpr
-    auto operator()(I a, T b) const noexcept { return EVE_DISPATCH_CALL(a, b); }
+    template<ordered_value I, floating_ordered_value T>
+    constexpr EVE_FORCEINLINE as_wide_as_t<T, I> operator()(I a, T b) const noexcept { return EVE_DISPATCH_CALL(a, b); }
```
Comment (DenisYaroshevskiy) [inline include/eve/module/special/regular/rising_factorial.hpp:20]: constexpr again

File: test/unit/module/math/expm1.cpp (hunk 2 above)
Comment (jfalcou) [reply test/unit/module/math/expm1.cpp:44]: This is an actual mistake

File: test/unit/module/core/ldexp.cpp (hunk 3 above)
Comment (jfalcou) [reply test/unit/module/core/ldexp.cpp:54]: It was redundant with the one below

File: include/eve/module/special/regular/rising_factorial.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/module/special/regular/rising_factorial.hpp:20]: As per discussion on slack, I am for supporting constexpr as it makes the scalar version usable in more scenario.

**Pull request merged**